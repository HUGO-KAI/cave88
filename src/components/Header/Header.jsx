import React, {useEffect, useState}from "react";
import { Link, useNavigate } from 'react-router-dom'
import style from './Header.module.scss'
import logo from '../../assets/logo.jpg'
//import banniere from '../../assets/banniere.webp'
import { BsFillTelephoneFill } from "react-icons/bs";
import {MdLocationOn} from "react-icons/md";
import Slideshow from "../Slideshow/Slideshow"

//Component Header
function Header (props) {
  const navigate = useNavigate();
  const token = sessionStorage.getItem("token");
  const [online,setOnline] = useState()
  const handleClick = () => {
    sessionStorage.clear("token")
    setOnline(false);
    navigate('/');
  }
  useEffect(()=>{
      if (!token) {
        setOnline(false)
      }else {
        setOnline(true)
      }
  },[token])
  return (
    <header>
      <div className ={`${style.limitedWidthBlockContainer} ${style.informations}`} >
        <div className = {style.limitedWidthBlock}>
          <ul>
            <li>
              <BsFillTelephoneFill size="14px" className= {style.informations__phone} />
              <a href = {"tel:"+props.tel}>{props.tel}</a>
            </li>
            <li><MdLocationOn size="16px" className={style.informations__address}/>{props.city}</li>
          </ul>
        </div>
      </div>
      <div className= {`${style.limitedWidthBlockContainer} ${style.menu}`} >
        <div className= {style.limitedWidthBlock} >
          <Link to="./">
            <img className={style.logo} src={logo} alt="Logo de l'entreprise" />
          </Link>
          <nav>
            <ul>
              <li><Link to="./">Accueil</Link></li>
              <li><Link to="./promo">Promo</Link></li>
            {!online?
              <li><Link to="./signIn">Sign in</Link></li>:
              <li className={style.signOut}><span onClick={handleClick}>Sign out</span></li>
            }  
            </ul>
          </nav>
        </div>
      </div>
      {
        /*<img className= {style.banniere} src= {banniere} alt="Baniere" />*/
      }
      <Slideshow />
    </header>
  )
}


export default Header