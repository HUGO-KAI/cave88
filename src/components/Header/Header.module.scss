/********/
/*Header*/
/********/
$color-primaire: #ad0036;
$color-secondaire: #ce134e;
$text-color: #3d4c68;

.limitedWidthBlockContainer {
	display: flex;
	justify-content: center;
}

.limitedWidthBlock {
	max-width: 1400px;
	box-sizing: border-box;
	margin: auto;
	width: 96%;
	padding-right:25px;
	display: flex;
	justify-content: space-between;
	@media (max-width: 575.98px) { 
		flex-direction: column;
		align-items: center;
		padding: 0px;
	}
}

/*informations*/
a{
	text-decoration: none;
	color:white
}

.informations {
	font-size: 13px;
	background-color: $color-secondaire;
}
.informations img {
	height: 17px;
}
.informations ul {
	display: flex;
	padding-left: 0;
	@media (max-width: 575.98px) { 
		flex-direction: column;
		align-items: center;
	}
}
.informations ul li {
	display: flex;
	list-style: none;
	margin-right: 36px;
	color:white;
	
	@media (max-width: 575.98px) { 
		margin-bottom: 14px;
		margin-right: 0;
	}
}
.informations__phone {
	padding-top: 3px;
	margin-right: 10px;
}
.informations__address {
	margin-right: 10px;
}

.logo {
	height: 100px;
	@media (max-width: 575.98px) { 
		margin-top: 20px;
	}
}

.menu{
	display: flex;
	justify-content: space-between;
	background-color: white;
	@media (max-width: 575.98px) {
		flex-direction: column;
		align-items: center;
	}
	nav {
		display: flex;
		float: right;
		@media (max-width: 575.98px) { 
			justify-content: space-around;
		}
		ul{
			display: flex;
			margin: 0;
			line-height: 130px;
			@media (max-width: 575.98px) {
				padding: 0;
				line-height: 80px;
				a:first-child li {
					margin-left: 0;
				}
			}
			li{
				list-style: none;
				margin-left: 50px;
				font-weight: 600;
				color: $text-color;
				a:hover {
					font-weight: 600;
					color: $color-primaire !important;
				}
				@media (max-width: 575.98px) {
					margin:0px 25px;
				}
			}
		} 
	}
	a {
		color:black;
		display: flex;
		align-items: center;
		text-decoration: none;
	}
	@media (max-width: 575.98px) { 
		nav ul {
			padding: 0;
			line-height: 80px;
		}
	}
}

/*banniere*/
.banniere {
	width: 100%;
	height: 300px;
	object-fit: cover;
	display: flex;
}

.signOut{
	cursor: pointer;
}

@media (max-width: 991.98px) {
	.informations ul {
		display: flex;
		justify-content: space-around;
	}
	.items a {
		width: 42%;
	}
	.footerMain .limitedWidthBlock {
		display: flex;
		flex-direction: column;
		align-items: center;
	}
	.footerSecondary .limitedWidthBlock p {
		text-align: center;
	}
}

