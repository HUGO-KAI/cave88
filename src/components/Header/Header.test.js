import {render, screen, cleanup} from "@testing-library/react";
import React from 'react'
import renderer from 'react-test-renderer'
import {BrowserRouter} from "react-router-dom"
import Header from "./Header";

afterEach (() => {
    cleanup()
}); 

test('should render Header component',()=>{
    render(
        //Utilise BrowerRouter pour enable useNavigate() hook
        <BrowserRouter>
            <Header/>
        </BrowserRouter>
    );
    const HeaderElement = screen.getByTestId('Header-1');
    expect(HeaderElement).toBeInTheDocument();
    expect(HeaderElement).toHaveTextContent('Accueil','Promo')
})

test('maches snapshot', ()=> {
    const telephone = '0642888868';
    const city = 'Aubervilliers';
    const tree = renderer.create(
        <BrowserRouter>
            <Header tel= {telephone} city = {city}/>
        </BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})