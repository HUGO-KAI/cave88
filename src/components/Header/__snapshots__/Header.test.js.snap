// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maches snapshot 1`] = `
<header
  data-testid="Header-1"
>
  <div
    className="limitedWidthBlockContainer informations"
  >
    <div
      className="limitedWidthBlock"
    >
      <ul>
        <li>
          <svg
            className="informations__phone"
            fill="currentColor"
            height="14px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.885.511a1.745 1.745 0 0 1 2.61.163L6.29 2.98c.329.423.445.974.315 1.494l-.547 2.19a.678.678 0 0 0 .178.643l2.457 2.457a.678.678 0 0 0 .644.178l2.189-.547a1.745 1.745 0 0 1 1.494.315l2.306 1.794c.829.645.905 1.87.163 2.611l-1.034 1.034c-.74.74-1.846 1.065-2.877.702a18.634 18.634 0 0 1-7.01-4.42 18.634 18.634 0 0 1-4.42-7.009c-.362-1.03-.037-2.137.703-2.877L1.885.511z"
              fillRule="evenodd"
            />
          </svg>
          <a
            href="tel:0642888868"
          >
            0642888868
          </a>
        </li>
        <li>
          <svg
            className="informations__address"
            fill="currentColor"
            height="16px"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5a2.5 2.5 0 010-5 2.5 2.5 0 010 5z"
            />
          </svg>
          Aubervilliers
        </li>
      </ul>
    </div>
  </div>
  <div
    className="limitedWidthBlockContainer menu"
  >
    <div
      className="limitedWidthBlock"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Logo de l'entreprise"
          className="logo"
          src="logo.jpg"
        />
      </a>
      <nav>
        <ul>
          <li>
            <a
              href="/"
              onClick={[Function]}
            >
              Accueil
            </a>
          </li>
          <li>
            <a
              href="/promo"
              onClick={[Function]}
            >
              Promo
            </a>
          </li>
          <li>
            <a
              href="/signIn"
              onClick={[Function]}
            >
              Sign in
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <div
    className="slide-container"
  >
    <div
      aria-roledescription="carousel"
      dir="ltr"
    >
      <div
        className="react-slideshow-container"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      >
        <button
          aria-label="Previous Slide"
          className="nav default-nav "
          data-type="prev"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M16.67 0l2.83 2.829-9.339 9.175 9.339 9.167-2.83 2.829-12.17-11.996z"
            />
          </svg>
        </button>
        <div
          className="react-slideshow-wrapper slide "
        >
          <div
            className="images-wrap"
            style={
              Object {
                "transform": "translate(-0px)",
              }
            }
          >
            <div
              aria-hidden="true"
              aria-roledescription="slide"
              data-index={-1}
            >
              <div>
                <div
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundImage": "url(https://cdn.pixabay.com/photo/2015/03/08/22/34/wine-664826_960_720.jpg)",
                      "backgroundSize": "cover",
                      "display": "flex",
                      "height": "400px",
                      "justifyContent": "center",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "background": "rgb(173 0 54 / 0.5)",
                        "color": "white",
                        "fontWeight": "bold",
                        "padding": "20px",
                      }
                    }
                  >
                    整箱优惠-Hard discount
                  </span>
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              aria-roledescription="slide"
              className="active"
              data-index={0}
            >
              <div>
                <div
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundImage": "url(https://cdn.pixabay.com/photo/2016/10/22/20/34/wines-1761613_1280.jpg)",
                      "backgroundSize": "cover",
                      "display": "flex",
                      "height": "400px",
                      "justifyContent": "center",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "background": "rgb(173 0 54 / 0.5)",
                        "color": "white",
                        "fontWeight": "bold",
                        "padding": "20px",
                      }
                    }
                  >
                    阿发酒庄-cave88vip
                  </span>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              aria-roledescription="slide"
              className=""
              data-index={1}
            >
              <div>
                <div
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundImage": "url(https://cdn.pixabay.com/photo/2015/03/08/22/35/wine-664832_960_720.jpg)",
                      "backgroundSize": "cover",
                      "display": "flex",
                      "height": "400px",
                      "justifyContent": "center",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "background": "rgb(173 0 54 / 0.5)",
                        "color": "white",
                        "fontWeight": "bold",
                        "padding": "20px",
                      }
                    }
                  >
                    批发零售-Prix grossiste
                  </span>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              aria-roledescription="slide"
              className=""
              data-index={2}
            >
              <div>
                <div
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundImage": "url(https://cdn.pixabay.com/photo/2015/03/08/22/34/wine-664826_960_720.jpg)",
                      "backgroundSize": "cover",
                      "display": "flex",
                      "height": "400px",
                      "justifyContent": "center",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "background": "rgb(173 0 54 / 0.5)",
                        "color": "white",
                        "fontWeight": "bold",
                        "padding": "20px",
                      }
                    }
                  >
                    整箱优惠-Hard discount
                  </span>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              aria-roledescription="slide"
              data-index={3}
            >
              <div>
                <div
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundImage": "url(https://cdn.pixabay.com/photo/2016/10/22/20/34/wines-1761613_1280.jpg)",
                      "backgroundSize": "cover",
                      "display": "flex",
                      "height": "400px",
                      "justifyContent": "center",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "background": "rgb(173 0 54 / 0.5)",
                        "color": "white",
                        "fontWeight": "bold",
                        "padding": "20px",
                      }
                    }
                  >
                    阿发酒庄-cave88vip
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          aria-label="Next Slide"
          className="nav default-nav "
          data-type="next"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M5 3l3.057-3 11.943 12-11.943 12-3.057-3 9-9z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</header>
`;
